pg_shard.c

```diff
@@ -179,6 +184,11 @@ _PG_init(void)
 							 &UseCitusDBSelectLogic, false, PGC_USERSET, 0, NULL,
 							 NULL, NULL);
 
+	DefineCustomBoolVariable("pg_shard.log_distributed_statements",
```

Comment:
The logic I used to get me `log_distributed_statements`

  * Some settings begin with `debug_` and others with `log_`, though the debug ones tend to be about logging a complicated structure, not one line. So I went with `log_`
  * The (probably well-known) configuration for PostgreSQL to log every query is `log_statement`. I therefore wanted to use _statement_ in my name
  * As an end-user I understand _query_ to mean a `SELECT`: something that asks for data and returns it. A _statement_, however, has side-effects
  * `distributed` is because these statements are fields in the distributed plan. I'm logging them at the point of planning, not execution, so they are not yet remote (`log_statement` also logs far before execution)
  * Finally, it's plural because unlike `log_statement`, we're going to have multiple statements per plan

So in short, I don't like _query_ (inappropriate for `INSERT`, `UPDATE`, or `DELETE`) and `remote` makes sense if I'm logging in the course of remote execution, which I'm not. Tell me if this isn't convincing.